baseplate/experiments/__init__.py

```diff
@@ -116,32 +126,36 @@ def variant(self, name, bucketing_event_override=None,
         :rtype: :py:class:`str`
         :return: Variant name if a variant is active, None otherwise.
         """
-        experiment_config = self._get_config(name)
-        if not experiment_config:
+        experiment = self._get_experiment(name)
+
+        if experiment is None:
             return None
 
-        experiment = parse_experiment(experiment_config)
-        variant = experiment.variant(**kwargs)
+        span_name = "{}.{}".format(self._context_name, "variant")
+        with self._span.make_child(span_name, local=True, component=name):
```

Comment:
I'll note, I'm not really sure that I'm naming this local span properly, in this case the span will look like `context_name.variant` and the value for LOCAL_COMPONENT will be `experiment_name`